### ISSUE TITLE
Freshly generated typescript project gives typescript warnings

### ISSUE THREAD

**Issue by patotoma** [honojs/hono#232]
I'm seeing some typescript warnings/errors when using middlewares and default typescript setup generated by wrangler:

```
No overload matches this call.
  Overload 1 of 2, '(path: string, ...middleware: Handler<string, Env>[]): Hono<Env, "">', gave the following error.
    Argument of type '(c: Context<string, Env>, next: Next) => Promise<void>' is not assignable to parameter of type 'Handler<string, Env>'.
      Types of parameters 'next' and 'next' are incompatible.
        Type 'Next | undefined' is not assignable to type 'Next'.
          Type 'undefined' is not assignable to type 'Next'.
  Overload 2 of 2, '(...middleware: Handler<string, Env>[]): Hono<Env, "">', gave the following error.
    Argument of type 'string' is not assignable to parameter of type 'Handler<string, Env>'.
```

```
Cannot invoke an object which is possibly 'undefined'.
```

<img width="375" alt="Screen Shot 2022-05-15 at 10 00 29 AM" src="https://user-images.githubusercontent.com/3104587/168455214-a14955ef-8b31-4588-a682-b24306deee12.png">

Please see full code at https://github.com/patotoma/hono-example

**Comment by yusukebe** [comment]
Hi @patotoma 

I've fixed the issue, and released v1.2.2. Please update Hono.

**Comment by patotoma** [comment]
No more warnings/errors in v1.2.2., thanks a lot!

Btw, loving Hono so far, amazing job!
